sktime/datatypes/_check.py

```diff
@@ -0,0 +1,187 @@
+# -*- coding: utf-8 -*-
+"""Machine type checkers for scitypes.
+
+Exports
+-------
+check_is(obj, mtype: str, scitype: str)
+    checks whether obj is mtype for scitype
+    returns boolean yes/no and metadata
+
+check_raise(obj, mtype:str, scitype:str)
+    checks whether obj is mtype for scitype
+    returns True if passes, otherwise raises error
+"""
+
+__author__ = ["fkiraly"]
+
+__all__ = [
+    "check_is",
+    "check_raise",
+    "mtype",
+]
+
+import numpy as np
+
+from sktime.datatypes._panel import check_dict_Panel
+from sktime.datatypes._series import check_dict_Series
+from sktime.datatypes._registry import mtype_to_scitype
+
+# pool convert_dict-s and infer_mtype_dict-s
+check_dict = dict()
+check_dict.update(check_dict_Series)
+check_dict.update(check_dict_Panel)
+
+
+def check_is(
+    obj, mtype: str, scitype: str = None, return_metadata=False, var_name="obj"
+):
+    """Check object for compliance with mtype specification, return metadata.
+
+    Parameters
+    ----------
+    obj - object to check
+    mtype: str or list of str, mtype to check obj as
+    scitype: str, optional, scitype to check obj as; default = inferred from mtype
+        if inferred from mtype, list elements of mtype need not have same scitype
+    return_metadata - bool, optional, default=False
+        if False, returns only "valid" return
+        if True, returns all three return objects
+    var_name: str, optional, default="obj" - name of input in error messages
+
+    Returns
+    -------
+    valid: bool - whether obj is a valid object of mtype/scitype
+    msg: str or list of str - error messages if object is not valid, otherwise None
+            str if mtype is str; list of len(mtype) with message per mtype if list
+            returned only if return_metadata is True
+    metadata: dict - metadata about obj if valid, otherwise None
+            returned only if return_metadata is True
+        fields:
+            "is_univariate": bool, True iff series has one variable
+            "is_equally_spaced": bool, True iff series index is equally spaced
```

Comment:
the refactor would then just set `return_metadata` = True, no?
I think it might be a bit cumbersome for the user if there are three return arguments?

There's two usage scenarios, one for users who want to check their input data, and one for internal use. I thought the metadata would only be useful in the second, so default behaviour (for the user) should be the "straigtforward expected return type".

Reply:
> the refactor would then just set `return_metadata` = True, no?

that would be the quick refactor. However, you would also be able to tidy up many functions and improve readability of the code, e.g. remove need to define the `ret` function in various functions.

> There's two usage scenarios, one for users who want to check their input data, and one for internal use. I thought the metadata would only be useful in the second, so default behaviour (for the user) should be the "straigtforward expected return type".

I cannot think of anything on the top of my head for how to deal with this. Hence why I think the current state is good for now, and we can refactor later.